gems/pending/spec/openstack/amqp/openstack_rabbit_event_monitor_spec.rb

```diff
@@ -48,4 +46,25 @@
     # get back to this piece
   end
 
+  context "remove legacy queues" do
+    before :each do
+    end
+
+    it "successfully removes legacy queues" do
+      @rabbit_channel = double
+      @rabbit_connection.stub(:create_channel).and_return(@rabbit_channel)
+      @rabbit_channel.stub(:close).and_return(nil)
+
+      @rabbit_connection.should_receive(:queue_exists?).at_least(2).times.with(/^miq-/).and_return(true)
+      @rabbit_channel.should_receive(:queue_delete).at_least(2).times.with(/^miq-/).and_return(nil)
+
+      @rabbit_connection.should_receive(:queue_exists?).once.with("notifications.*").and_return(true)
+      @rabbit_channel.should_receive(:queue_delete).once.with("notifications.*").and_return(nil)
+
+      @event_monitor = OpenstackRabbitEventMonitor.new(@options)
+
+      @event_monitor.send(:remove_legacy_queues)
```

Comment:
testing a private method, but I figured the complexity around testing the public method that calls this was a little too much for a single test .. and since it was a stupid mistake in this method that bit me before, testing it directly seemed like a decent idea